lib/src/soloud.dart

```diff
@@ -328,24 +324,33 @@ class SoLoud {
 
   /// Load a new sound to be played once or multiple times later
   ///
+  /// [loadIntoMem] if true Soloud::wav will be used which loads
+  /// all audio data into memory. This will be useful when
+  /// the audio is short, ie for game sounds, mainly used to prevent
+  /// gaps or lags when starting a sound (less CPU, more memory allocated).
+  /// If false, Soloud::wavStream will be used and the audio data is loaded 
+  /// from the given file when needed (more CPU, less memory allocated).
+  /// See the [seek] note problem when using [loadIntoMem] = false
+  /// Default is true.
   /// [completeFileName] the complete file path
   /// Returns PlayerErrors.noError if success and a new sound
   ///
   Future<({PlayerErrors error, SoundProps? sound})> loadFile(
-    String completeFileName,
-  ) async {
+    String completeFileName, {
+    bool loadIntoMem = true,
```

Comment:
The nit about naming this `loadIntoMemory` applies here more than anywhere else — this is the developer-facing API, if I'm not mistaken.

Reply:
Actually, one other thing that comes to mind is to make this an enum?

```dart
enum LoadMode {
  /// Load and decompress the audio file into RAM.
  memory,

  /// Keep the file on disk and only load chunks as needed.
  disk,
}
```

Then we'd have:

```dart
loadFile('path/to/music.mp3', mode: LoadMode.disk);
```

The enum also gives us an easier path later, if we decide to add more options. For example, if we add something like `LoadMode.balanced` or whatever, the developer-facing API stays the same.

To be clear: on the inside, this should still be implemented with a `bool`. At least for now. It's just about what the developer uses.